introduction_and_learning/Histogram/histogram_api.py

```diff
@@ -0,0 +1,90 @@
+__all__ = [
+    "HistogramInterface"
+]
+from abc import abstractmethod, ABCMeta
+
+
+class HistogramInterface(metaclass=ABCMeta):
+    """
+    HistogramInterface is basic interface for creating histogram from various
+    file types classes.
+
+    HistogramInterface is abstract class. Methods that required are
+        read_data(file_path)
+        create(data_input)
+        generate_report(histogram, output_type, path)
+    """
+
+    @abstractmethod
+    def read_data(self, file_path: str) -> str:
+        """
+        This method handling opening and reading from given file. Type of this
+        file is given in class definition.
+
+        Parameters
+        ----------
+            file_path: string, required
+                Localisation of file which data should be read from.
+
+        Returns
+        -------
+        Method return before read data on success or information
+        "File does not have [file extension like .txt] extension." on
+        failure.
+
+        Exaples
+        -------
+        >>> histogram_txt.read_data("myDoc/file.txt")
+        >>> histogram_json.read_data("myDoc/file.json")
+        """
+        raise NotImplementedError
```

Comment:
Here I was a little bit confused beceause I do not where to write docstring and I decided to write it here beceause all classes and method based on this abstract class will be do same things. But in the other hand writing docstrings in inherited class could make code more readable. Which option is better? or docstring should be here in abstract class and in inherited class.

Reply:
Overall, yes, you can define a method body in abstract class method and this is a common approach when you have the same implementation ;) no issues here.

About docstring, you can inherit a docstring from a parent class like this:
`child.__doc__ = parent.__doc__` for example in class constructor or even in the class body directly.